### DISCUSSION TITLE
CSV Data

### DISCUSSION THREAD

**Discussion by ghost** [gitfrosh/lotr-api#52] [General]
I am currently working on refreshing/adding to the .csv files.   As you could imagine, it's a daunting task, as I am cross-referencing two different websites, adding new data to each line, and correcting punctuation and whitespace errors.
I am curious as to what method was used initially to retrieve all the data.  I'm wondering if there's a way to implement automation for this with a web scraper or something... or is this something that I will just have to be patient with and work on for the next few months?  lol
Eager to hear what you all have to say!

**Comment by ghost** [comment]
Here are all the columns I'm working with now:

`name, aliases, titles, rule, race, birth, death, gender, culture, languages, hair, height, realm, spouse, children, weapon, wikiUrl, imageUrl`

**Comment by ghost** [comment]
As of now, I am working on implementing a web scraper that will check both references, extrapolate, then consolidate the info, followed by adding that data to the CSV.  Not entirely sure if I will be able to pull this off, but so far so good.  I've managed to set up web scraping on one website successfully.

**Comment by ghost** [comment]
Successfully implemented web scraping on both sites.  Currently trying to implement functionality to cross-reference the retrieved data and consolidate into single object.

**Comment by ghost** [comment]
Due to poor construction of the second source I was using, it is extremely difficult to correctly scrape data for each character, as some pages are different from others, and none of the the elements have IDs.  For now, I am going to forego the second source, as this is above my skill level, if possible at all.

**Comment by gitfrosh** [comment]
Hi @ghost, I’m assuming this discussion can be closed since we haven’t heard back for a while.
Thanks for all your efforts and contributions!
